src/backend/app.ts

```diff
@@ -0,0 +1,142 @@
+
+  constructor(routes: Routes[]) {
+    this.app = express()
+    this.port = process.env.port ? parseInt(process.env.port) : 8080
+
+    this.initializeMiddlewares()
+    if (process.env.ENABLE_AUTH === 'true') {
+      console.log(
+        'OAuth via PassportJS is enabled because ENABLE_AUTH is set to "true"'
+      )
+      this.initializeLogin()
+    } else {
+      console.log(
+        'OAuth via PassportJS is disabled because ENABLE_AUTH is not set to "true"'
+      )
+    }
+    this.initializeRoutes(routes)
+    this.initializeRenderEngine()
+  }
+
+  public listen(): void {
+    const server = http.createServer(this.app)
+    server.keepAliveTimeout = 1000 * (60 * 6) // 6 minutes
+    server.listen(this.port, () => {
+      console.log(`🚀 App listening on the port ${this.port}`)
+    })
+  }
+
+  public getServer(): express.Application {
+    return this.app
+  }
+
+  private initializeMiddlewares() {
+    this.app.use(cors())
+    this.app.use(express.static('./src/public'))
+    this.app.use(bodyParser.urlencoded({ extended: true }))
+    this.app.use(bodyParser.json())
+  }
+
+  private initializeRoutes(routes: Routes[]) {
+    routes.forEach((route) => {
+      this.app.use('/', route.router)
+    })
+  }
+
+  private initializeRenderEngine() {
+    nunjucks.configure(views, {
+      autoescape: true,
+      express: this.app,
+    })
+    this.app.engine('html', nunjucks.render)
+    this.app.set('views', views)
+    this.app.set('view engine', 'html')
+  }
+
+  private initializeLogin() {
+    passport.serializeUser((user: any, done: any) => done(null, user))
+    passport.deserializeUser((user: any, done: any) => done(null, user))
+
+    this.app.use(
+      session({
+        secret: 'keyboard cat',
+        resave: false,
+        saveUninitialized: false,
+        cookie: { secure: true },
+        store: redisStore,
+      })
+    )
+
+    this.app.set('trust proxy', 1) // trust first proxy
+    this.app.use(passport.initialize())
+    this.app.use(passport.session())
+
+    passport.use(
+      new OAuth2Strategy(
+        {
+          authorizationURL: process.env.OAUTH_AUTH_URL || 'not set',
+          tokenURL: process.env.OAUTH_TOKEN_URL || 'not set',
+          clientID: process.env.OAUTH_ID || 'not set',
+          clientSecret: process.env.OAUTH_SECRET || 'not set',
+          callbackURL: process.env.OAUTH_CALLBACK_URL || 'not set',
+        },
+        function (
+          accessToken: string,
+          refreshToken: string,
+          profile: any,
+          cb: any
+        ) {
+          console.log(
+            'OAuth2Strategy callback',
+            accessToken,
+            refreshToken,
+            profile
+          )
+          cb(null, profile)
+        }
+      )
+    )
+
+    this.app.get('/login', passport.authenticate('oauth2'))
+    this.app.get(
+      '/auth/gds/callback',
+      passport.authenticate('oauth2', '/error-callback'),
+      async (req, res) => res.redirect('/')
+    )
```

Comment:
With these being technically routes, how about having a `LoginRoute` or `AuthRoute` class?
The /reauth endpoint will be there and will need a controller.

Reply:
Happy to address this change when rebasing my PR on this one

Reply:
Happy for you do address this as your PR is to do with Auth